app/src/main/java/net/gsantner/markor/activity/DocumentEditAndViewFragment.java

```diff
@@ -291,12 +293,6 @@ public void onPause() {
     @Override
     public void onSaveInstanceState(@NonNull Bundle outState) {
         outState.putSerializable(SAVESTATE_DOCUMENT, _document);
-
-        // Don't save contents - will be reloaded from file
-        _hlEditor.setHighlightingEnabled(false);
-        _hlEditor.setAutoFormatEnabled(false);
-        _hlEditor.setText("");
```

Comment:
onSaveInstance state is called all the time without a corresponding restore. This causes the file contents to be lost basically every time the fragment is paused